argoverse/utils/sim2.py

```diff
@@ -68,16 +86,29 @@ def scale(self) -> float:
 
     @property
     def matrix(self) -> np.ndarray:
-        """Calculate 3*3 matrix group equivalent"""
+        """Calculate 3*3 matrix group equivalent."""
         T = np.zeros((3, 3))
         T[:2, :2] = self.R_
         T[:2, 2] = self.t_
         T[2, 2] = 1 / self.s_
         return T
 
     def compose(self, S: "Sim2") -> "Sim2":
-        """Composition with another Sim2."""
-        return Sim2(self.R_ * S.R_, ((1.0 / S.s_) * self.t_) + self.R_ @ S.t_, self.s_ * S.s_)
+        """Composition with another Sim2.
+
+        This can be understood via block matrix multiplication, if self is parameterized as (R1,t1,s1)
+        and if `S` is parameterized as (R2,t2,s2):
+
+        [R1  t1]   [R2  t2]   [R1 @ R2   R1@t2 + t1/s2]
+        [0 1/s1] @ [0 1/s2] = [ 0          1/(s1*s2)  ]
+        """
+        # fmt: off
```

Comment:
Is there a benefit of enabling/disabling the formatting vs. new variables?

Reply:
@benjaminrwilson so I tried to think of a new variable name that would be clear, but it seemed to just add more complexity and more things to understand.

I think forcing the code onto 3 different lines avoids creating an unnecessary temporary variable and makes it easier to read.